test/api-test.js

```diff
@@ -270,6 +270,34 @@ describe('RippleAPI', function () {
       }, /dropsToXrp:\ invalid\ value\ '\.\.\.'\,\ should\ be\ a\ number\ matching\ \(\^\-\?\[0\-9\]\*\.\?\[0\-9\]\*\$\)\./)
     })
   })
+
+  describe('deriveKeypair', function () {
+    it('returns keypair for secret', function () {
+      var keypair = this.api.deriveKeypair("snsakdSrZSLkYpCXxfRkS4Sh96PMK");
+      assert(keypair.privateKey == "008850736302221AFD59FF9CA1A29D4975F491D726249302EE48A3078A8934D335");
+      assert(keypair.publicKey == "035332FBA71D705BD5D97014A833BE2BBB25BEFCD3506198E14AFEA241B98C2D06");
+    })
+
+    it("throws with an invalid secret", function (){
+      assert.throws(() => {
+        this.api.deriveKeypair("...");
+      }, "Non-base58 character")
```

Comment:
You must use a regex here for the comparison to actually work. If you change this to `"Non-base59 character"`, the test still passes, despite the `9`.

Here's a regex for this one:
```
/^Error\: Non\-base58 character$/
```

See https://nodejs.org/api/assert.html#assert_assert_throws_block_error_message

> Due to the confusing notation, it is recommended not to use a string as the second argument. This might lead to difficult-to-spot errors.